### PR TITLE
Implement exclude repo by name and pattern for ADO

### PR DESCRIPTION
**Pull request by varsanojidan**
Closes https://github.com/sourcegraph/sourcegraph/issues/46267
Added the ability to exclude repo by name and pattern.

## Test plan
Added to unit test and verified manually.

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
LGTM! :shipit:

File: schema/azuredevops.schema.json [hunk 1]
```diff
@@ -40,6 +40,33 @@
...
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "object",
+        "title": "ExcludedAzureDevOpsServerRepo",
+        "additionalProperties": false,
+        "anyOf": [{ "required": ["name"] }, { "required": ["id"] }, { "required": ["pattern"] }],
+        "properties": {
+          "name": {
+            "description": "The name of an Azure DevOps Services/Server repo (\"projectName/repositoryName\") to exclude from mirroring.",
```
Comment (mrnugget) [inline schema/azuredevops.schema.json:55]:
```suggestion
            "description": "The name of an Azure DevOps Services/Server project and repository (\"projectName/repositoryName\") to exclude from mirroring.",
```

**Review by mrnugget — APPROVED** [review]

File: schema/azuredevops.schema.json (hunk 1 above)
Comment (varsanojidan) [reply schema/azuredevops.schema.json:55]: Sounds good.

I am also going to push the change you requested from the last PR here.
image

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+0.78 kb) | 0.01% (+0.78 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/iv/ado-exclude-repos/compare-report.html;tab=live_object) for a full comparison between the commits f405356 and a37627c or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: schema/azuredevops.schema.json (hunk 1 above)
Comment (mrnugget) [reply schema/azuredevops.schema.json]: Beautiful!

**Pull request merged**